monix-catnap/shared/src/main/scala/monix/catnap/internal/AsyncUtils.scala

```diff
@@ -17,28 +17,20 @@
 
 package monix.catnap.internal
 
-import cats.implicits._
-import cats.effect.{Async, ExitCase}
-import monix.catnap.FutureLift
-import monix.execution.Callback
-import scala.concurrent.Promise
+import cats.effect.{Async}
 
 private[monix] object AsyncUtils {
   /**
-    * Describing the `cancelable` builder for any `Async` data type,
-    * in terms of `bracket`.
+    * The `cancelable` builder from cats-effect 2
     */
   def cancelable[F[_], A](k: (Either[Throwable, A] => Unit) => F[Unit])(implicit F: Async[F]): F[A] =
-    F.asyncF { cb =>
-      val p = Promise[A]()
-      val awaitPut = Callback.fromPromise(p)
-      val future = p.future
-      val futureF = FutureLift.scalaToAsync(F.pure(future)).attempt.map(cb)
-      val cancel = k(awaitPut)
+    F.async[A] { cb => F.pure(Some(k(cb))) }
```

Comment:
`F.cancelable` and `F.asyncF` have been rolled into one `F.async`. Old `F.async` became `F.async_`. I made the helpers to avoid larger source changes.